app/models/notification.rb

```diff
@@ -172,4 +174,12 @@ def subject_number
   def display_thread?
     Octobox.include_comments? && subjectable? && subject.present? && user.display_comments?
   end
+
+  def github_client
+    if app_installation.present?
+      user.app_installation_client
```

Comment:
So here I am using `app_installation_client`. This matches most closely what was happening here: https://github.com/octobox/octobox/pull/1262/files#diff-cf94d05b646cd6dc6622869c79a2c4dbL31

Where it would use this token if the App was installed for the User. 

I call this out becuase it is different from the `#github_client` method that we have on `Subject`.
I think the root of my question is, What is the difference between 

```
Octokit::Client.new(access_token: user.app_token, auto_paginate: true)
```

and

```
Octobox.installation_client(self.github_id)
```


I _think_ that this usage here, using `user.app_installation_client` is correct! Since this is trying to fetch User scoped data, so I think we need a user scoped token. The `installation_client` seems to be in a User-less context.
So if that is correct, I am not sure if that means the usage in `Subject` is correct...

Reply:
Correct, the `user.app_token` is only different from a token generated for the App in that it has more context around the user making the request, which in this case doesn't matter.